From b171836c67c7151483da10eb7d5af03918531539 Mon Sep 17 00:00:00 2001
From: Sanskar Jain <sansjain@qti.qualcomm.com>
Date: Thu, 16 Oct 2025 10:50:53 +0530
Subject: [PATCH 04/40] qcacld-3.0: Assign NULL to memory location after
 freeing it

After freeing the memory for *entries array it is not assigned
to NULL. This can lead to use_after_free of memory and can
cause segmentation fault.

Assign *entries pointer to NULL after freeing the memory for it.

CRs-Fixed: 4315444
Change-Id: Iec98523186a406b75aef65ce7bcd26f4063daeab
---
 components/wmi/src/wmi_unified_roam_tlv.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/components/wmi/src/wmi_unified_roam_tlv.c b/components/wmi/src/wmi_unified_roam_tlv.c
index 23d105ebab..9270abe01b 100644
--- a/components/wmi/src/wmi_unified_roam_tlv.c
+++ b/components/wmi/src/wmi_unified_roam_tlv.c
@@ -4207,6 +4207,7 @@ free_entries:
 	qdf_mem_zero(*entries,
 		     WLAN_MAX_ML_BSS_LINKS * sizeof(**entries));
 	qdf_mem_free(*entries);
+	*entries = NULL;
 
 free_keys:
 	for (k = 0; k < WMI_NUM_KEYS_ALLOCATED; k++) {
-- 
2.34.1

